packages/dds/tree/src/test/simple-tree/api/schemaFactory.spec.ts

```diff
@@ -381,6 +381,26 @@ describe("schemaFactory", () => {
 			assert.deepEqual(schema.fields.get("bar")!.metadata, barMetadata);
 		});
 
+		it("Node schema metadata", () => {
+			const factory = new SchemaFactory("");
+
+			class Foo extends factory.object(
+				"Foo",
+				{ bar: factory.number },
+				{ metadata: { description: "An object", custom: { baz: true } } },
+			) {}
+
+			assert.equal(Foo.metadata?.description, "An object");
+			assert.equal(Foo.metadata?.custom?.baz, true);
+
+			const foo = hydrate(Foo, { bar: 37 });
+			const schema = Tree.schema(foo) as ObjectNodeSchema;
+
+			assert.equal(schema.metadata?.description, "An object");
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			assert.equal((schema.metadata?.custom as any)?.baz, true);
```

Comment:
I though the point of the extra type parameter was to not have to do casts like this. Is this test then showing its not typing correctly?

Reply:
Oh, I see you casted away the type.

I think a less verbose and stricter check would be `assert.equal(Tree.schema(foo), Foo)`

Given we have other tests for that, I don't think the bottom of this test is necessary.

Also note that Tree.schema works fine on unhydrated nodes, so the call to hydrate is not needed even if you keep this part of the test